### ISSUE TITLE
Client is not receiving updates

### ISSUE THREAD

**Issue by HanaGemela** [owncloud/client#7591]
macOS 10.15
client 2.6.0 (build 12703)

Steps to recreate:
1) Change the update channel to get the update

Actual result:
This happened after update from 2.5.4 to 2.6.0. Last time this behaviour was seen, the cause was whitespace in updater yml config

![image](https://user-images.githubusercontent.com/49001702/68788348-7ee90f80-0643-11ea-9da8-967d55ea8dba.png)

Nothing in logs (client is not trying to reach the updater url):
``` 
11-13 12:15:11:089 [ debug gui.updater ]	[ -[DelegateObject updaterMayCheckForUpdates:] ]: may check: YES
11-13 12:15:11:428 [ debug gui.updater ]	[ -[DelegateObject updater:didFinishLoadingAppcast:] ]:	
11-13 12:15:11:440 [ debug gui.updater ]	[ -[DelegateObject updaterDidNotFindUpdate:] ]:	
```

Expected result: Message that the client is up to date

**Comment by jnweiger** [comment]
This error is seen with one mac after updating to 2.6.0 with another mac after updating to 260rc4.   
We cannot see a similar error messages on windows or Ubuntu with 2.6.0

Online updater for mac stopped, until we know more. If it is an issue with the client, users may lose the ability to get further updates. 

@guruz @ckamm @dschmidt @TheOneRing please investigate.

**Comment by jnweiger** [comment]
Further investigation:

* The error message appears also on 2.6.0 RC2
* The error message goes away, when the updater is changed to actually offer a newer update.

2.6.0.12703 is the version number that was just installed on mac. 
Switch to beta channel triggers the error message. (Beta channel has 2.6.0.11xxx, which is lower)
Switch back to stable channel again tiriggers the same error (Stable channel has 2.6.0.12703, which is identical)
Change the updater to advertise version 2.0.6.12704
Switch back and forth multiple times. Repeats the error messages multiple times, until eventually the updater (due to throtteling) announces this new version. The client recognizes the update without error, and suggests download and install.

My conclusion so far: The updater is correct, it must be something in the macos specific updater code, that broke before 2.6.0rc2

**Comment by guruz** [comment]
> My conclusion so far: The updater is correct, it must be something in the macos specific updater code, that broke before 2.6.0rc2

I don't see anything suspicious in git log -p ./src/gui/updater/

@dschmidt @TheOneRing Did you change the Sparkle version we ship with?

**Comment by guruz** [comment]
I can reproduce it too by changing (in rc4) from beta to normal channel.
```
11-14 11:06:09:359 [ debug gui.updater ]	[ -[DelegateObject updaterMayCheckForUpdates:] ]:	may check: YES
11-14 11:06:09:829 [ debug gui.updater ]	[ -[DelegateObject updater:didFinishLoadingAppcast:] ]:	
11-14 11:06:09:847 [ debug gui.updater ]	[ -[DelegateObject updaterDidNotFindUpdate:] ]:	

```

**Comment by guruz** [comment]
But actually I don't think that beta->stable switch is a normal bug, the original reason posted here by @HanaGemela is a bug!

> This happened after update from 2.5.4 to 2.6.0.

**Comment by guruz** [comment]
I don't see anything immediately in the Console.app that could explain this:
```
standard	11:11:58.059815 +0100	owncloud	Task <C9A44374-C857-4622-BD24-D5C8E2E38751>.<1> received response, status 200 content U
standard	11:11:58.059993 +0100	owncloud	Task <C9A44374-C857-4622-BD24-D5C8E2E38751>.<1> response ended
standard	11:11:58.062333 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:58.085179 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:58.098298 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:58.126418 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:58.128950 +0100	EmojiFunctionRowIM_Extension	IMKTextInputFunctionRowItemServiceViewController -dealloc *invoked* on NSViewService (<IMKTextInputFunctionRowItemServiceViewController: 0x60c0000d23d0>)
standard	11:11:58.129165 +0100	EmojiFunctionRowIM_Extension	IMKTextInputFunctionRowItemServiceViewController -dealloc *invoked* on NSViewService (<IMKTextInputFunctionRowItemServiceViewController: 0x60c0000d6810>)
standard	11:11:58.155935 +0100	owncloud	TIC TCP Conn Cancel [3:0x7fcab55c23d0]
standard	11:11:58.156200 +0100	owncloud	[3 <private> stream, pid: 4167, url: https://updates.owncloud.com/client/?version=2.6.0.12665&platform=macos&oem=ownCloud&versionsuffix=rc4&channel=beta&sparkle=true, tls] cancelled
	[3.1 37269867-1169-4426-BEF5-F086F7D869AB <private>.59532<-><private>]
	Connected Path: satisfied (Path is satisfied), interface: en0, ipv4, dns
	Duration: 0.399s, DNS @0.000s took 0.001s, TCP @0.002s took 0.064s, TLS took 0.166s
	bytes in/out: 4148/770, packets in/out: 7/8, rtt: 0.070s, retransmitted packets: 0, out-of-order packets: 0
standard	11:11:58.156483 +0100	owncloud	TIC TCP Conn Destroyed [3:0x7fcab55c23d0]
fehler	11:11:58.156858 +0100	owncloud	TIC Read Status [3:0x0]: 1:57
fehler	11:11:58.156887 +0100	owncloud	TIC Read Status [3:0x0]: 1:57
standard	11:11:58.162184 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:59.409616 +0100	mdworker	MacOS error: -67050
standard	11:11:59.411018 +0100	mdworker	MacOS error: -67050
standard	11:11:59.426547 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:59.445504 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:59.461134 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:11:59.463685 +0100	mdworker	MacOS error: -67050
standard	11:11:59.465230 +0100	mdworker	MacOS error: -67050
standard	11:11:59.476385 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:12:00.032843 +0100	trustd	could not enable test hierarchy: no UAT pinning preferences set
standard	11:12:00.032847 +0100	trustd	could not disable pinning: not an internal release
standard	11:12:00.651522 +0100	EmojiFunctionRowIM_Extension	IMKTextInputFunctionRowItemServiceViewController -dealloc *invoked* on NSViewService (<IMKTextInputFunctionRowItemServiceViewController: 0x6040000d1410>)
standard	11:12:00.656136 +0100	EmojiFunctionRowIM_Extension	IMKTextInputFunctionRowItemServiceViewController -dealloc *invoked* on NSViewService (<IMKTextInputFunctionRowItemServiceViewController: 0x6080000d40b0>)
standard	11:12:00.964962 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:12:00.978783 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:12:00.993466 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
standard	11:12:01.211408 +0100	trustd	cert[2]: AnchorTrusted =(leaf)[force]> 0
```

Is ```MacOS error: -67050``` from us? mnhm

**Comment by jnweiger** [comment]
Latest sparkle is confused, when 
* the item element is missing in the response rss
* the item.enclose element has no url, no sparkle:version or no sparkle:dsaSignature attribute.

Suggested fix:
When throttling, we return a minimal item with properly built enclose sub-element, where url and sparkle:dsaSignature are empty strings, and sparkle:version has the version from the request.

**Comment by guruz** [comment]
@jnweiger FYI don't put "fixes #123456" or "fixes $URL" into a commit message, github would automatically close it

**Comment by HanaGemela** [comment]
Client 2.6.0 (build 12703), macOS 10.15 shows 'You're up-to-date' message
